index.ts

```diff
@@ -80,71 +76,72 @@ function safeClosest(event: Event, selector: string): Element | void {
 	}
 }
 
+type TReturnType<TOptionsType extends undefined | boolean | AddEventListenerOptions> = TOptionsType extends {signal: AbortSignal} ? undefined : AbortController;
+
 /**
  * Delegates event to a selector.
  * @param options A boolean value setting options.capture or an options object of type AddEventListenerOptions
  */
 function delegate<
 	Selector extends string,
 	TElement extends Element = ParseSelector<Selector, HTMLElement>,
-	TEventType extends EventType = EventType
+	TEventType extends EventType = EventType,
+	TOptionsType extends undefined | boolean | AddEventListenerOptions = undefined
 >(
 	base: EventTarget | Document | ArrayLike<Element> | string,
 	selector: Selector,
 	type: TEventType,
 	callback: delegate.EventHandler<GlobalEventHandlersEventMap[TEventType], TElement>,
-	options?: boolean | AddEventListenerOptions
-): delegate.Subscription;
+	options?: TOptionsType
+): TReturnType<TOptionsType>;
 
 function delegate<
 	TElement extends Element = HTMLElement,
-	TEventType extends EventType = EventType
+	TEventType extends EventType = EventType,
+	TOptionsType extends undefined | boolean | AddEventListenerOptions = undefined
 >(
 	base: EventTarget | Document | ArrayLike<Element> | string,
 	selector: string,
 	type: TEventType,
 	callback: delegate.EventHandler<GlobalEventHandlersEventMap[TEventType], TElement>,
-	options?: boolean | AddEventListenerOptions
-): delegate.Subscription;
+	options?: TOptionsType
+): TReturnType<TOptionsType>;
 
 // This type isn't exported as a declaration, so it needs to be duplicated above
 function delegate<
 	TElement extends Element,
-	TEventType extends EventType = EventType
+	TEventType extends EventType = EventType,
+	TOptionsType extends undefined | boolean | AddEventListenerOptions = undefined
 >(
 	base: EventTarget | Document | ArrayLike<Element> | string,
 	selector: string,
 	type: TEventType,
 	callback: delegate.EventHandler<GlobalEventHandlersEventMap[TEventType], TElement>,
-	options?: boolean | AddEventListenerOptions
-): delegate.Subscription {
+	options?: TOptionsType
+): TReturnType<TOptionsType> {
+	const listenerOptions: AddEventListenerOptions = typeof options === 'object' ? options : {capture: options};
```

Comment:
Let's also drop the problematic `once` option while we're at it:
```suggestion
	const listenerOptions: AddEventListenerOptions = typeof options === 'object' ? options : {capture: options};

	delete listenerOptions.once
```

For `once` to be supported, it would have to be handled by `delegate-it`, never passed to the native `addEventListener`. This is a bug I just noticed.

Reply:
"Handling" it requires extra code. I'd rather merge this first